src/interfaces/wallet.cpp

```diff
@@ -244,17 +244,33 @@ class WalletImpl : public Wallet
         bool sign,
         int& change_pos,
         CAmount& fee,
-        std::string& fail_reason) override
+        std::string& fail_reason,
+        CMWTx mwtx) override
```

Comment:
nit: We should probably pass CMWTx around by ref. Can you change this to `const CMWTx& mwtx`? Same with `Wallet::CreateTransaction`.

Right now, I know it just contains a shared_ptr, but that may not always be the case.